src/UKHO.ERPFacade.Common/IO/Azure/AzureTableReaderWriter.cs

```diff
@@ -379,6 +405,35 @@ await foreach (var entity in entities)
             return records.FirstOrDefault();
         }
 
+        public async Task UpdateStatusOfRecordOfSaleEntityPublisedEvent(string correlationId)
+        {
+            TableClient tableClient = GetTableClient(RecordOfSaleTableName);
+            RecordOfSaleEventEntity existingEntity = await GetRecordOfSaleEventEntity(correlationId);
+
+            if (existingEntity != null)
+            {
+                existingEntity.Status = "Complete";
+                await tableClient.UpdateEntityAsync(existingEntity, ETag.All, TableUpdateMode.Replace);
+
+                _logger.LogInformation(EventIds.UpdatedStatusOfRecordOfSalePublishedEventInAzureTable.ToEventId(), "Status of existing record of sale published event updated in azure table successfully.");
+            }
+        }
+
+        public async Task UpdateStatusOfUploadedLicenceRecordOfSaleEntityEvent(string correlationId)
```

Comment:
Please rename the method name as per event name like UpdateStatusOfLicenceUpdatedPublishedEvent

Reply:
Renamed the method name as suggested above.